exonum-java-proofs/src/main/java/com/exonum/binding/storage/proofs/map/flat/CheckedMapProof.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.storage.proofs.map.flat;
+
+import com.exonum.binding.hash.HashCode;
+import java.util.List;
+
+/**
+ * A checked map proof.
+ * In case of incorrect proof all methods (except for getStatus and compareWithRootHash)
+ * throw IllegalStateException.
+ * Example usage:
+ * <pre><code>
+ * byte[] key = "The key for which I want a proved value";
+ * HashCode expectedRootHash = // get a known root hash from block proof //
+ * UncheckedMapProof proof = requestProofForKey(key);
+ * // Convert to checked
+ * CheckedMapProof checkedProof = proof.check();
+ * // Check the root hash
+ * if (checkedProof.compareWithRootHash(HashCode expectedRootHash)) {
```

Comment:
`checkedProof.compareWithRootHash(HashCode expectedRootHash)` does not work in Java :-)

Reply:
`checkedProof.compareWithRootHash(expectedRootHash)`?

Reply:
sorry, missed that)